internal/api/http/middleware/access_interceptor.go

```diff
@@ -81,46 +81,69 @@ func (a *AccessInterceptor) DeleteExhaustedCookie(writer http.ResponseWriter) {
 	a.cookieHandler.DeleteCookie(writer, a.limitConfig.ExhaustedCookieKey)
 }
 
+func (a *AccessInterceptor) HandleIgnorePathPrefixes(ignoredPathPrefixes []string) func(next http.Handler) http.Handler {
+	return a.handle(ignoredPathPrefixes...)
+}
+
 func (a *AccessInterceptor) Handle(next http.Handler) http.Handler {
-	if !a.svc.Enabled() {
-		return next
-	}
-	return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
-		ctx := request.Context()
-		tracingCtx, checkSpan := tracing.NewNamedSpan(ctx, "checkAccess")
-		wrappedWriter := &statusRecorder{ResponseWriter: writer, status: 0}
-		limited := a.Limit(tracingCtx)
-		checkSpan.End()
-		if limited {
-			a.SetExhaustedCookie(wrappedWriter, request)
-			http.Error(wrappedWriter, "quota for authenticated requests is exhausted", http.StatusTooManyRequests)
-		}
-		if !limited && !a.storeOnly {
-			a.DeleteExhaustedCookie(wrappedWriter)
-		}
-		if !limited {
-			next.ServeHTTP(wrappedWriter, request)
-		}
-		tracingCtx, writeSpan := tracing.NewNamedSpan(tracingCtx, "writeAccess")
-		defer writeSpan.End()
-		requestURL := request.RequestURI
-		unescapedURL, err := url.QueryUnescape(requestURL)
-		if err != nil {
-			logging.WithError(err).WithField("url", requestURL).Warning("failed to unescape request url")
+	return a.handle()(next)
+}
+
+func (a *AccessInterceptor) handle(ignoredPathPrefixes ...string) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		if !a.svc.Enabled() {
+			return next
 		}
-		instance := authz.GetInstance(tracingCtx)
-		a.svc.Handle(tracingCtx, &access.Record{
-			LogDate:         time.Now(),
-			Protocol:        access.HTTP,
-			RequestURL:      unescapedURL,
-			ResponseStatus:  uint32(wrappedWriter.status),
-			RequestHeaders:  request.Header,
-			ResponseHeaders: writer.Header(),
-			InstanceID:      instance.InstanceID(),
-			ProjectID:       instance.ProjectID(),
-			RequestedDomain: instance.RequestedDomain(),
-			RequestedHost:   instance.RequestedHost(),
+		return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			ctx := request.Context()
+			tracingCtx, checkSpan := tracing.NewNamedSpan(ctx, "checkAccess")
+			wrappedWriter := &statusRecorder{ResponseWriter: writer, status: 0}
+			for _, ignoredPathPrefix := range ignoredPathPrefixes {
+				if strings.HasPrefix(request.RequestURI, ignoredPathPrefix) {
+					checkSpan.End()
+					next.ServeHTTP(wrappedWriter, request)
+					a.writeLog(tracingCtx, wrappedWriter, writer, request, true)
+					return
+				}
```

Comment:
wouldn't this change the behaviour? i want that any ignored path directly is passed to the next handler without checking for the limit and if none matches it will do the limitation check
your suggestion would always ignore the limit

Reply:
ah i could replace `continue` with `break`, but unsure whether that's more readable

Reply:
Why should it break the behaviour? currently if the the prefix does not match the forloop also continues